tachyon/crypto/random/xor_shift/xor_shift_rng.h

```diff
@@ -29,38 +28,28 @@ class XORShiftRNG final : public RNG {
 
   XORShiftRNG() = default;
 
-  template <typename Container>
-  static XORShiftRNG FromSeed(const Container& seed) {
-    CHECK_EQ(std::size(seed), kSeedSize);
-    static_assert(std::is_same_v<base::container_value_t<Container>, uint8_t>,
-                  "The value type of |seed| must be uint8_t");
-    XORShiftRNG ret;
-    memcpy(&ret.x_, &seed[0], sizeof(uint32_t));
-    memcpy(&ret.y_, &seed[4], sizeof(uint32_t));
-    memcpy(&ret.z_, &seed[8], sizeof(uint32_t));
-    memcpy(&ret.w_, &seed[12], sizeof(uint32_t));
-    return ret;
-  }
-
-  static XORShiftRNG FromSeed(const uint8_t seed[kSeedSize]) {
-    XORShiftRNG ret;
-    memcpy(&ret.x_, &seed[0], sizeof(uint32_t));
-    memcpy(&ret.y_, &seed[4], sizeof(uint32_t));
-    memcpy(&ret.z_, &seed[8], sizeof(uint32_t));
-    memcpy(&ret.w_, &seed[12], sizeof(uint32_t));
-    return ret;
-  }
-
-  static XORShiftRNG FromRandomSeed() {
+  // RNG methods
+  void SetRandomSeed() override {
     uint8_t seed[kSeedSize];
     uint64_t lo = base::Uniform(base::Range<uint64_t>::All());
     uint64_t hi = base::Uniform(base::Range<uint64_t>::All());
     memcpy(&seed[0], &lo, sizeof(uint64_t));
     memcpy(&seed[8], &hi, sizeof(uint64_t));
```

Comment:
```suggestion
    *reinterpret_cast<uint64_t*>(&seed[0]) =
        base::Uniform(base::Range<uint64_t>::All());
    *reinterpret_cast<uint64_t*>(&seed[8]) =
        base::Uniform(base::Range<uint64_t>::All());
```

It is minor(optional) but we can avoid `memcpy` with something like this.

Reply:
Feel free to resolve once you read.

Reply:
For curiosity I checked and the asm difference disappears from `-O1` https://godbolt.org/z/K481zYasf